File: src/main/java/ua/softgroup/matrix/server/service/UserService.java
package ua.softgroup.matrix.server.service;

import ua.softgroup.matrix.api.model.datamodels.AuthModel;
import ua.softgroup.matrix.server.persistent.entity.User;

import java.util.Optional;

public interface UserService extends GeneralEntityService<User> {

    String authenticate(AuthModel authModel);

    Optional<User> getByUsername(String username);

    Optional<User> getByTrackerToken(String token);

}

File: src/main/scala/ua/softgroup/matrix/server/service/UserService.scala
package ua.softgroup.matrix.server.service

import java.util.Optional

import ua.softgroup.matrix.api.model.datamodels.AuthModel
import ua.softgroup.matrix.server.persistent.entity.User

/**
  * @author Oleksandr Tyshkovets <sg.olexander@gmail.com> 
  */
trait UserService extends GenericEntityService[User] {

  def authenticate(authModel: AuthModel): Option[String]

  def getByUsername(username: String): Option[User]

  def getByTrackerToken(token: String): Optional[User]

}
